commit 0e6086a7de36e539d54eb82e61cf94e7f5363c02
Author:     David Vrabel <[hidden email]>
AuthorDate: Thu Apr 4 17:21:12 2013 +0000
Commit:     Ian Jackson <[hidden email]>
CommitDate: Mon Apr 8 17:05:46 2013 +0100

    xl: default autoballoon option to "auto"
   
    In xl.conf, autoballoon="auto" will do the right thing for most
    people.  Make it the default (instead of "on").
   
    Signed-off-by: David Vrabel <[hidden email]>
    Committed-by: Ian Jackson <[hidden email]>
---
 docs/man/xl.conf.pod.5 |    2 +-
 tools/libxl/xl.c       |    6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/docs/man/xl.conf.pod.5 b/docs/man/xl.conf.pod.5
index a4ce3e5..aaf8da1 100644
--- a/docs/man/xl.conf.pod.5
+++ b/docs/man/xl.conf.pod.5
@@ -60,7 +60,7 @@ You are strongly recommended to set this to C<"off"> (or C<"auto">) if
 you use the C<dom0_mem> hypervisor command line to reduce the amount
 of memory given to domain 0 by default.
 
-Default: C<"on">
+Default: C<"auto">
 
 =item B<run_hotplug_scripts=BOOLEAN>
 
diff --git a/tools/libxl/xl.c b/tools/libxl/xl.c
index ac41fcd..16cd3f3 100644
--- a/tools/libxl/xl.c
+++ b/tools/libxl/xl.c
@@ -38,7 +38,7 @@
 xentoollog_logger_stdiostream *logger;
 int dryrun_only;
 int force_execution;
-int autoballoon = 1;
+int autoballoon = -1;
 char *blkdev_start;
 int run_hotplug_scripts = 1;
 char *lockfile;
@@ -99,10 +99,12 @@ static void parse_global_config(const char *configfile,
         else if (!strcmp(buf, "off") || !strcmp(buf, "0"))
             autoballoon = 0;
         else if (!strcmp(buf, "auto"))
-            autoballoon = auto_autoballoon();
+            autoballoon = -1;
         else
             fprintf(stderr, "invalid autoballoon option");
     }
+    if (autoballoon == -1)
+        autoballoon = auto_autoballoon();
 
     if (!xlu_cfg_get_long (config, "run_hotplug_scripts", &l, 0))
         run_hotplug_scripts = l;
--
generated by git-patchbot for /home/xen/git/xen.git#master
